### PR TITLE
Small performance improvements for GeomBuilder._buildCoords

### PR DESCRIPTION
**Pull request by groutr**
Some minor improvements to help build coordinate lists a little more efficiently.

1. Lift the ndim check out of the loop. This only needs to be checked once per call.
2. Build tuples directly instead of creating a list and converting to tuples.

### CODE REVIEW

File: fiona/_geometry.pyx [hunk 1]
```diff
@@ -124,15 +124,20 @@ cdef class GeomBuilder:
     cdef list _buildCoords(self, void *geom):
         # Build a coordinate sequence
         cdef int i
+        cdef int npoints
+        cdef list coords
```
Comment (sgillies) [inline fiona/_geometry.pyx:128]:
```suggestion
        cdef list coords = []
```

And then delete line 133?

**Review by sgillies — APPROVED** [review]
@groutr thank you! I'll backport to maint-1.10 after it's merged.

I made one suggestion.

Would a list comprehension be any faster?

File: fiona/_geometry.pyx (hunk 1 above)
Comment (groutr) [reply fiona/_geometry.pyx:128]: Seems fine to me.

**Comment by groutr** [comment]
I would be shocked if a list comprehension performed any different from a for-loop since they both generate identical C code.

**Review by sgillies — APPROVED** [review]

**Comment by sgillies** [comment]
Back ported to maint-1.10.

**Pull request merged**